[WIP] io-stats.c: re-organize stat structure

mykaul opened the pull request:
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
io-stats.c: re-organize stat structure

Updates: #1863
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/afr-no-fsync.t

0 test(s) generated core 


4 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/basic/afr/afr-no-fsync.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2656/

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/afr-no-fsync.t

0 test(s) generated core 


5 test(s) needed retry 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/basic/afr/afr-no-fsync.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2685/

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/afr-no-fsync.t

0 test(s) generated core 


5 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/basic/afr/afr-no-fsync.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2696/

mykaul commented:
tests/basic/afr/afr-no-fsync.t passes for me on my Fedora 36 on this branch.

Commit:
afr-no-fsync.t: disable cluster.lookup-optimize (since the volume is sharded), filter FSYNCDIR

Per https://github.com/gluster/glusterfs/issues/2253 we need to disable lookup-optimize feature.

More importantly, let xlators write their own directories under .glusterfs, before profiling and looking for FSYNC.
(They do an FSYNCDIR as part of the mkdir() of creating the dirs)

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression

mykaul commented:
> tests/basic/afr/afr-no-fsync.t passes for me on my Fedora 36 on this branch.

Disabled lookup-optimize in afr-no-fsync.t test - let's see if it makes a difference.

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/afr-no-fsync.t

0 test(s) generated core 


4 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/basic/afr/afr-no-fsync.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2698/

mykaul commented:
> > tests/basic/afr/afr-no-fsync.t passes for me on my Fedora 36 on this branch.
> 
> Disabled lookup-optimize in afr-no-fsync.t test - let's see if it makes a difference.

Additionally, filter FSYNCDIR (we are looking I believe for FSYNC, not FSYNCDIR)

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/afr-read-hash-mode.t

0 test(s) generated core 


4 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/basic/afr/afr-read-hash-mode.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2699/

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/afr-read-hash-mode.t

0 test(s) generated core 


4 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/basic/afr/afr-read-hash-mode.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2700/
